### DISCUSSION TITLE
Assertion that an event has occurred

### DISCUSSION THREAD

**Discussion by webtrees-pesz** [FamilySearch/GEDCOM#258] [Polls]
In the current version of GEDCOM, an event is considered to have occurred if either a date, a place, or a payload specification such as Y is specified. In my opinion, this rule should be extended if an AGE substructure is also available. The AGE substructure describes the age of a person when an event occurred.

Best regards
Peter

**Comment by kiwi3685** [comment]
I believe it is generally accepted that age at death is one of, if not actually the, most inaccurate of any individual's vital records. The reason is simple, it is the one record the person concerned cannot provide. So it must come from 3rd party, and therefore is less reliable. It should not be given the same importance as date or place of death, which can be readily verified.

**Comment by tychonievich** [comment]
I find the decision to pick some details as an assertion of happening and others not to be peculiar, but at this point I think changing it would cause more problems than otherwise. For roughly 25 years files have been generated where something like

```gedcom
0 @I1@ INDI
1 NAME Qiyuan /Wu/
1 NATU
2 AGE > 32y
```

means "we're not sure a naturalization happened, but if it did happened then Qiyuan was at least 32 years old when it did." Should that meaning have been defined? I'm not convinced. But given that it was, changing its meaning now would also change the meaning of old files, making them assert more than they intended.

If we want to normalize the "event has occurred" data, the only way I know of that we could do so where old files can be losslessly converted to the new format would be to remove the special case for DATE and PLAC: if we made such a change then we could convert all old files with DATE or PLAC to new files with an additional Y at no loss of information. But while I like the uniformity of that change, I don't think it is worth the disruption to current tools and corresponding probability of implementation errors it would create.

**Reply by elyoh** [reply]
Ambiguity is already baked in because GEDCOM 5.5.1 restricts the use of the `Y` flag to just 4 events.  `INDIVIDUAL_EVENT_STRUCTURE` allows its use with `BIRT`, `CHR`, and `DEAT` events whilst `FAMILY_EVENT_STRUCTURE` allows it only with `MARR` events.  This was a change from GEDCOM 5.5 and the GEDCOM 5.5 behaviour was restored for GEDCOM 7.0.

In the example cited, the `Y` assertion flag would never have been permitted on a `NATU` event in GEDCOM 5.5.1.

**Reply by elyoh** [reply]
Moreover, there is a limitation with allowing presence of a `DATE` or `PLAC` tag assert an event happened.  When an event includes an indeterminate date such as `AFT JAN 2023` the event is asserted to occur and there is no way to indicate otherwise.

Consider an equivalent representation using an indeterminate `AGE`:

"We're not sure a death happened, but if it did happened then John was at least 32 years old when it did." 
```
0 @I1@ INDI
1 NAME John /Smith/
1 BIRT
2 DATE 2 JAN 1990
1 DEAT
2 AGE > 32y
```

"We're sure a death happened, and John was at least 32 years old when it did." 
```
0 @I1@ INDI
1 NAME John /Smith/
1 BIRT
2 DATE 2 JAN 1990
1 DEAT Y
2 AGE > 32y
```
If using a `DEAT` tag, both the following mean:
"We're sure a death happened, and John was at least 32 years old when it did." 
```
0 @I1@ INDI
1 NAME John /Smith/
1 DEAT
2 DATE AFT JAN 2023
```
```
0 @I1@ INDI
1 NAME John /Smith/
1 DEAT Y
2 DATE AFT JAN 2023
```

Maximum expressibility is only possible if the assertion that an event occured is made with the `Y` flag only.

**Comment by Norwegian-Sardines** [comment]
Saying something happened should be based on a source.  Therefore, having a source_structure should also indicate that an event occurred, or an attribute can be affirmed.  Would it not be better to assert an event happened or attribute was true if and only if a source was attached?   

On the other hand!
With regard to AGE, or any other information found but not dated/placed, if all I’m doing is collecting information from sources and have not yet made a conclusion, would it be wrong to still enter an AGE (or other detail) for an event or attribute without knowing the place or date?  If we find that a individual was Adopted, but have no date, place or age, but yet a credible source was found that stated this fact, it should be entered and at least be considered true unless some other source refutes that claim and proves to be more reliable?

**Reply by elyoh** [reply]
My view is that decisions about whether it is right or wrong to record indeterminate / unevidenced data is for the author not the specification.  A 'conclusion' as to whether an event occurred is a judgement made on the part of an author according to their own surety criteria, and is for the author, not the specification to determine.  Consequently, it should be stored separately and should not be inferred from presence of specific tags, such as `DATE`, `PLAC`, or even `SOUR`.  Such an inference is only possible when there is a constraint on those tags that mean they may only contain information which assert an event occurred.  GEDCOM does not impose such a constraint.  It allows the `Y` flag in empty events i.e. assertion can be made without any included evidence. `DATE` payloads can be indeterminate and `SOUR` payloads may include evidence of when an event did not happen, and therefore presence of these alone should not be used to assert something happened.
